scripts/files/tests/fs_test.py

```diff
@@ -89,6 +89,21 @@ def test_write_sidecars_one_found(capsys: CaptureFixture[str], subtests: SubTest
     rmtree(target)
 
 
+def test_write_all_in_order(setup: str) -> None:
+    inputs: list[str] = []
+    i = 0
+    while i < 10:
+        path = Path(os.path.join(setup, str(i)))
+        if i % 2 == 0:
+            path.write_text("a" * 1000 * 1000, encoding="utf-8")  # 1MB
```

Comment:
NIT: would be good to generating a bunch of garbage once rather than generating it every time